core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -482,6 +517,18 @@ class Partition(val topic: String,
     laggingReplicas
   }
 
+  def appendRecordsToFutureReplica(records: MemoryRecords) {
+    getReplica(Request.FutureLocalReplicaId).get.log.get.appendAsFollower(records)
+  }
+
+  def appendRecordsToFollower(records: MemoryRecords) {
+    // The read lock is needed to avoid race condition while ReplicaAlterDirThread is executing
+    // maybeDeleteAndSwapFutureReplica() to replace follower replica with the future replica
```

Comment:
How about changing the comment slightly to the following?

"The read lock is needed to prevent the follower replica from being updated while ReplicaAlterDirThread is executing maybeDeleteAndSwapFutureReplica() to replace follower replica with the future replica."

Reply:
Sure. I have updated the comment as suggested.